include/sparrow_ipc/arrow_interface/arrow_array_schema_common_release.hpp

```diff
@@ -0,0 +1,62 @@
+
+#pragma once
+
+#include <sparrow/c_interface.hpp>
+
+#include "arrow_array/private_data.hpp"
+#include "arrow_schema/private_data.hpp"
+
+namespace sparrow_ipc
+{
+    /**
+     * Release the children and dictionnary of an `ArrowArray` or `ArrowSchema`.
+     *
+     * @tparam T `ArrowArray` or `ArrowSchema`
+     * @param t The `ArrowArray` or `ArrowSchema` to release.
+     */
+    template <class T>
+        requires std::same_as<T, ArrowArray> || std::same_as<T, ArrowSchema>
+    void release_common_non_owning_arrow(T& t)
+    {
+        using private_data_type = std::conditional_t<
+            std::same_as<T, ArrowArray>,
+            non_owning_arrow_array_private_data,
+            non_owning_arrow_schema_private_data>;
+        if (t.release == nullptr)
+        {
+            return;
+        }
+        SPARROW_ASSERT_TRUE(t.private_data != nullptr);
+        const auto private_data = static_cast<const private_data_type*>(t.private_data);
```

Comment:
The private data is allocated in `fill_non_owning_arrow_schema`, so I would expect this function to deallocate it?

Reply:
It's deallocated in the release_non_owning_arrow_array and release_non_owning_arrow_schema function, but you are right, let's move it to the common releaser